app/controllers/coronavirus_form/additional_product_controller.rb

```diff
@@ -35,7 +35,7 @@ def update_session_store
 
   def previous_path
     session["product_details"] ||= []
-    latest_product_id = (session["product_details"].last || {}).dig("product_id")
+    latest_product_id = (session["product_details"].last || {}).with_indifferent_access.dig("product_id")
```

Comment:
I know you didn't add it, but rather than using a "OR", the safe navigation operator could be used:

```
latest_product_id = session["product_details"]&.with_indifferent_access.dig("product_id")
```

Reply:
Thanks!